internal/cmd/keys.go

```diff
@@ -42,19 +44,22 @@ func newKeysAddCmd() *cobra.Command {
 	var options keyCreateOptions
 
 	cmd := &cobra.Command{
-		Use:   "add KEY IDENTITY",
+		Use:   "add IDENTITY",
 		Short: "Create an access key",
 		Long:  `Create an access key. Only machine identities are supported at this time.`,
 		Example: `
 # Create an access key named 'example-key' that expires in 12 hours
 $ infra keys add example-key machine-a --ttl=12h
 `,
-		Args: ExactArgs(2),
+		Args: ExactArgs(1),
 		RunE: func(cmd *cobra.Command, args []string) error {
-			keyName := args[0]
-			machineName := args[1]
+			machineName := args[0]
 
-			if strings.Contains(keyName, " ") {
+			if options.Name == "" {
+				options.Name = fmt.Sprintf("%s-%s", machineName, time.Now().Format("20060102150405.000000"))
```

Comment:
I was hoping for names in the format of <adjective>-<noun> :D

Reply:
I'm not a fan of pet names. It might be more reader friendly but it's not very descriptive. This format was picked because it conveys some information to the user if they elect to not give the key a more descriptive name